### PR TITLE
[docs] Add LightGBM-Ray docs, update XGBoost-Ray docs

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

Adds documentation for LightGBM-Ray (as well as all of the necessary changes to requirements and code). Updates XGBoost-Ray docs to match the current release and adds scikit-learn API docs.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -107,10 +108,74 @@ def import_or_mock(module):
...
+]
 for xgb_mod in xgb_modules:
     import_or_mock(xgb_mod)
 
+# replace refs in XGBoost documentation XGBoost-Ray inherits from
+sys.modules["xgboost_ray"].RayXGBClassifier.fit.__doc__ = sys.modules[
+    "xgboost_ray"].RayXGBClassifier.fit.__doc__.replace(
+        ":ref:`callback_api`", "Callback API")
+sys.modules["xgboost_ray"].RayXGBRegressor.fit.__doc__ = sys.modules[
+    "xgboost_ray"].RayXGBRegressor.fit.__doc__.replace(":ref:`callback_api`",
```
Comment (richardliaw) [inline doc/source/conf.py:123]: Hmm, could you give some context for all of this?

File: doc/source/conf.py [hunk 2]
```diff
@@ -107,10 +108,74 @@ def import_or_mock(module):
...
+    "lightgbm_ray"].RayLGBMClassifier.predict_proba.__doc__ = sys.modules[
+        "lightgbm_ray"].RayLGBMClassifier.predict_proba.__doc__.replace(
+            "RayParams", "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].RayLGBMRegressor.fit.__doc__ = sys.modules[
+    "lightgbm_ray"].RayLGBMRegressor.fit.__doc__.replace(
+        "RayParams", "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].RayLGBMRegressor.predict.__doc__ = sys.modules[
+    "lightgbm_ray"].RayLGBMRegressor.predict.__doc__.replace(
+        "RayParams", "lightgbm_ray.RayParams")
+
```
Comment (krfricke) [inline doc/source/conf.py:178]: This clutters the config a bit. Can we do something like this

```
replaces = [
     (sys.modules["lightgbm_ray"].RayLGBMRegressor.fit, "RayParams", "lightgbm_ray.RayParams"),
     # ....
]
for m, s, t in replaces:
    m.__doc__ = m.__doc__.replace(s,t)
```

and add documentation before the block why this is needed and wht it achieves?

Comment (krfricke) [reply doc/source/conf.py:178]: or maybe even leave out the last two items of the tuple

```
replaces = [
     sys.modules["lightgbm_ray"].RayLGBMRegressor.fit,
    sys.modules["lightgbm_ray"].RayLGBMRegressor.predict,
     # ....
]
for m in replaces:
    m.__doc__ = m.__doc__.replace("RayParams", "lightgbm_ray.RayParams")
```

**Review by krfricke — APPROVED** [review]
Thanks!

**Pull request merged**